### PR TITLE
Profiler: Run perf on HF binaries, insert counter between operations

### PR DESCRIPTION
**Pull request by davidchuyaya**

### CODE REVIEW

File: hydroflow_plus/src/snapshots/hydroflow_plus__profiler__tests__predicate_pushdown_through_map.snap.new [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (shadaj) [inline hydroflow_plus/src/snapshots/hydroflow_plus__profiler__tests__predicate_pushdown_through_map.snap.new:1]: you prob need to regenerate snapshots: `cargo insta test -p hydroflow_plus` and then `cargo insta review`

you may need to `cargo install cargo-insta --locked` if the command is not found

File: hydro_deploy/core/src/hydroflow_crate/mod.rs [hunk 2]
```diff
@@ -78,6 +80,11 @@ impl HydroflowCrate {
         self
     }
 
+    pub fn perf(mut self) -> Self {
+        self.perf = true;
+        self
+    }
+
```
Comment (MingweiSamuel) [inline hydro_deploy/core/src/hydroflow_crate/mod.rs:87]: Might want to make this
```rust
pub fn perf(mut self, perf: bool) -> Self {
    self.perf = perf;
    self
}
```
so you can turn it off too

File: hydro_deploy/core/src/ssh.rs [hunk 3]
```diff
@@ -305,6 +306,7 @@ impl<T: LaunchedSSHHost> LaunchedHost for T {
                 channel
                     .exec(&format!("{binary_path_string}{args_string}"))
                     .await?;
+                // TODO: If perf == true, launch SSH with perf
```
Comment (MingweiSamuel) [inline hydro_deploy/core/src/ssh.rs:309]: add `let _ = perf` to quell warning

File: hydroflow_lang/src/graph/hydroflow_graph.rs [hunk 4]
```diff
@@ -1519,6 +1531,9 @@ pub struct WriteConfig {
     /// Op text will only be their name instead of the whole source.
     #[cfg_attr(feature = "debugging", arg(long))]
     pub op_short_text: bool,
+    /// Op text will exclude any line that starts with "use".
+    #[cfg_attr(feature = "debugging", arg(long))]
+    pub op_text_no_imports: bool,
```
Comment (MingweiSamuel) [inline hydroflow_lang/src/graph/hydroflow_graph.rs:1536]: Seems a bit overly specific? Whats this for?

File: hydroflow_plus/src/profiler.rs
Comment (MingweiSamuel) [inline hydroflow_plus/src/profiler.rs:1]: Nit: use `///` for doc comments above `fn`s

File: hydroflow_plus/src/profiler.rs [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        let runtime_context = flow.runtime_context();
+        let built = flow.extract();
+
+        insta::assert_debug_snapshot!(&built.ir);
+
+        // Print mermaid
+        // let mut mermaid_config = WriteConfig {op_text_no_imports: true, ..Default::default()};
+        // for (_, ir) in built.clone().optimize_default().hydroflow_ir() {
+        //     println!("{}", ir.to_mermaid(&mermaid_config));
+        // }
```
Comment (MingweiSamuel) [inline hydroflow_plus/src/profiler.rs:103]: ^

File: hydroflow_lang/src/graph/hydroflow_graph.rs (hunk 4 above)
Comment (davidchuyaya) [reply hydroflow_lang/src/graph/hydroflow_graph.rs:1536]: This was to remove the import statements that was included in each block in the mermaid graph. Is there a better way to do it?

Comment (MingweiSamuel) [reply hydroflow_lang/src/graph/hydroflow_graph.rs:1536]: Does hydroflow+ generate a lot of `use`es?

Comment (davidchuyaya) [reply hydroflow_lang/src/graph/hydroflow_graph.rs:1536]: When I ran mermaid over the example in profiler.rs (commented out) I got a `use` for every node yeah

**Pull request merged**